handler.go

```diff
@@ -74,6 +74,30 @@ func (m *multiHandler) Flush() {
 	}
 }
 
+// FilteredHandler constructs a Handler that processes Measures with `filter` before forwarding to `h`
+func FilteredHandler(h Handler, filter func([]Measure) []Measure) Handler {
```

Comment:
Another API here would be `func(m Measure) bool`, but maybe you can't do that because you need the ability to compare multiple Measure ments. 

I'm nervous that this API might involve a lot of copying/allocations.

Reply:
Ok, so I imagine a for loop inside of `filter` that allocates a new slice (on the stack) and builds it from the input slice:
```
result []Measure = {}
range _, measure := measures {
  if CONDITION {
    result = append(measure, result)
  }
}
return result
```

I can move that loop up into HandleMeasures, which would look like:
```
filteredMeasures []Measure = {}
range _, measure := measures {
  if filter(measure) {
    filteredMeasures = append(filteredMeasures, result)
  }
}
return h.handler.HandleMeasures(time, filteredMeasures...)
```

It's almost the same code... just a slight change on where the function boundaries are. Given that the code is working as is, I'd rather leave it. Were you thinking of a different implementation? Is there a way to optimize away that stack slice allocation and copy that I don't see?

Reply:
No that's fine, you're right, this is probably thinking too hard